library/std/src/backtrace.rs

```diff
@@ -251,10 +252,17 @@ impl Backtrace {
                 Err(_) => false,
             },
         };
-        ENABLED.store(enabled as usize + 1, SeqCst);
+        Self::set_enabled(enabled);
         enabled
     }
 
+    /// Explicitly enable or disable backtrace capturing, overriding the default
+    /// derived from the `RUST_BACKTRACE` or `RUST_LIB_BACKTRACE` environment
+    /// variables.
+    pub fn set_enabled(enabled: bool) {
```

Comment:
Let's put an `unstable` attribute on this, since we've got #72981 open to stabilize the core backtrace API.